Content.Server/GameObjects/Components/Items/Storage/EntityStorageComponent.cs

```diff
@@ -361,6 +361,11 @@ bool IInteractUsing.InteractUsing(InteractUsingEventArgs eventArgs)
 
             if (!tool.UseTool(eventArgs.User, Owner, ToolQuality.Welding, 1f))
                 return false;
+            if(_contents.Contains(eventArgs.User))
```

Comment:
You should move this above the tool check up there.

Reply:
I went ahead and moved it so that it would be above the WelderComponent out line too so the component doesn't have to be checked. Should be a little faster without having to do both checks.